cpp/include/Ice/Protocol.h

```diff
@@ -23,39 +23,39 @@ namespace IceInternal
 // Compression status (std::uint8_t)
 // Message size (Int)
 //
-const ::std::int32_t headerSize = 14;
+const std::int32_t headerSize = 14;
 
 //
 // The magic number at the front of each message
 //
-extern const ::std::uint8_t magic[4];
+extern const std::byte magic[4];
 
 //
 // The current Ice protocol, protocol encoding and encoding version
 //
-const ::std::uint8_t protocolMajor = 1;
-const ::std::uint8_t protocolMinor = 0;
-const ::std::uint8_t protocolEncodingMajor = 1;
-const ::std::uint8_t protocolEncodingMinor = 0;
+const std::uint8_t protocolMajor = 1;
```

Comment:
Will we update these to use std::byte too?

Reply:
I went back and forth on these. Since they represent a actual version number and are used as such (sometimes also used as the byte for a version number) I think they probably make more sense left as a "numeric type"

Reply:
Yes, the idea is:
 - single byte => uint8_t
 - byte sequence => std::byte array/vector/pair of pointers/span